code/game/objects/items/weapons/implants/types/greytide_implant.dm

```diff
@@ -57,14 +45,39 @@
 		if(istype(I, /obj/item/weapon/implant/traitor) || istype(I, /obj/item/weapon/implant/loyalty))
 			if(I.imp_in == imp_in)
 				imp_in.visible_message("<span class='big danger'>[imp_in] seems to resist the implant!</span>", "<span class='danger'>You feel a strange sensation in your head that quickly dissipates.</span>")
+				imp_in.visible_message("<span class='danger'>The implant drops to the floor.</span>", "<span class='danger'>Your body rejects the implant, which unceremoniously drops to the floor.</span>")
 				remove(imp_in)
 				return
+
 	if(istraitor(imp_in) || ischallenger(imp_in))
 		imp_in.visible_message("<span class='big danger'>[imp_in] seems to resist the implant!</span>", "<span class='danger'>You feel a familiar sensation in your head that quickly dissipates.</span>")
+		imp_in.visible_message("<span class='danger'>The implant drops to the floor.</span>", "<span class='danger'>Your body rejects the implant, which unceremoniously drops to the floor.</span>")
 		remove(imp_in)
 		return
+
+	var/datum/faction/F = find_active_faction_by_typeandmember(/datum/faction/syndicate/greytide, null, implanter.mind)
+	if(!F)
+		message_admins("ERROR: Greytide implant failed to find valid faction at implanted() after passing faction check in insert().")
+		stack_trace("ERROR: Greytide implant failed to find valid faction at implanted() after passing faction check in insert().")
+		return
+	var/success = F.HandleRecruitedMind(imp_in.mind)
+	if(success)
+		F.forgeObjectives()
+		update_faction_icons()
+		var/datum/role/R = imp_in.mind.GetRole(IMPLANTSLAVE)
+		if(R)
+			R.AnnounceObjectives()
+
+	else /* This will never actually happen at present. I don't know why it's here, but it's worth keeping around in case anyone has a funny idea for it. */
+		visible_message("<span class = 'warning'>The head of [imp_in] begins to glow a deep red. It is going to explode!</span>")
+		spawn(3 SECONDS)
+			var/datum/organ/external/head/head_organ = imp_in.get_organ(LIMB_HEAD)
+			head_organ.explode()
+			qdel(src)
+		return
+
 	to_chat(imp_in, "<span class='notice'>You feel a surge of loyalty towards [implanter.name].</span>")
-	to_chat(imp_in, "<B><span class='big warning'>You've been shown the Greytide by [implanter.name]!</B> You now must lay down your life to protect them and assist in their goals at any cost.</span>")
+	to_chat(imp_in, "<span class='big warning'><b>You've been brainwashed into the Greytide by [implanter.name]!</b><br>You now must lay down your life to protect them and assist in his/her goals at any cost.</span>")
```

Comment:
Isn't this gramatically inconsistent? If we accept singular them, then singular their is a correct usage and this is an unncessary change. If we don't accept singular them, then his/her is incorrect as it's singular.

Reply:
fixed